backend/src/main/java/de/neuefische/backend/controller/ItemController.java

```diff
@@ -0,0 +1,41 @@
+package de.neuefische.backend.controller;
+
+import de.neuefische.backend.model.Item;
+import de.neuefische.backend.service.ItemService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/items")
+public class ItemController {
+    private final ItemService itemService;
+
+    @PostMapping
+    public Item create(@RequestBody Item item) {
+        return itemService.create(item);
+    }
+
+    @GetMapping
+    public List<Item> getAll() {
+        return itemService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public Item getById(@PathVariable String id) throws Exception {
+        return itemService.getById(id);
+    }
+
+    @PutMapping("/{id}")
+    public Item update(@PathVariable String id, @RequestBody Item item) {
+        item.setId(id);
```

Comment:
Let me just double check: wouldn't you be overwriting the MongoDB Id here? Is that deliberate? I would have expected a "standard" update endpoint to search for the id handed into it in the DB and than update the item this search retrieves...

Reply:
Guter Hinweis. Muss ich mir noch mal durch den kopf gehen lassen...